odoo/addons/base/tests/test_res_partner.py

```diff
@@ -309,3 +309,22 @@ def test_display_address_missing_key(self):
         )
         self.env["res.country"].invalidate_model()
         self.assertEqual(before, partner._display_address().strip())
+
+    def test_accessibility_of_company_partner_from_branch(self):
+        """ Check accessibility of company partner from branch. """
+        company = self.env['res.company'].create({'name': 'company'})
+        branch = self.env['res.company'].create({
+            'name': 'branch',
+            'parent_id': company.id
+        })
+        partner = self.env['res.partner'].create({
+            'name': 'partner',
+            'company_id': company.id
+        })
+        user = self.env['res.users'].create({
+            'name': 'user',
+            'login': 'user',
+            'company_id': branch.id,
+            'company_ids': [branch.id]
+        })
+        self.assertEqual(partner.with_user(user).check_access_rule('read'), None)
```

Comment:
This test is testing that a partner of a company is accessible to a user that works in a branch of this company. When it is the case, check_access_rule returns None, this is what I am checking. It seemed to do exactly what I need. 
I can change the case if you think checking what a tree returns is clearer.

Reply:
Functionally speaking what you change is the records people can access. check_access_rule is an implementation detail, what matters is what is changing in partner structure holding branches. So that is why I would test as it is clearer what is the reasoning behind the change.

Reply:
Ok I understand, I will update it. Thank you